File: client/src/domain/booking.js
export default class Booking {
  constructor(id = null, userId = null, userName = null, seatId = null, bookingDateTime = null) {
    this.id = id;
    this.userId = userId;
    this.userName = userName;
    this.seatId = seatId;
    this.bookingDateTime = new Date(bookingDateTime);
  }
}

export  class Room {
    constructor(id = null, name = null, seats = []) {
        this.id = id;
        this.name = name;
        this.seats = seats.map(seat => new Seat(seat.id, seat.roomId, seat.isAvailable, seat.bookings));
    }
}

export class Seat {
    constructor(id = null, roomId = null, isAvailable = true, bookings = []) {
        this.id = id;
        this.roomId = roomId;
        this.isAvailable = isAvailable;
        this.bookings = bookings;
    }
}

export class CreateBookingRequest {
  constructor(seatId, bookingDateTime, reCAPTCHAToken) {
    this.seatId = seatId;
    this.bookingDateTime = bookingDateTime.toISOString();
    this.reCAPTCHAToken = reCAPTCHAToken;
  }
}

export class DeleteBookingRequest {
  constructor(bookingId) {
    this.bookingId = bookingId
  }
}

File: server/DAL/Model/Booking.cs
namespace server.DAL.Models
{
    public class Booking
    {
        public int Id { get; set; }
        public string UserId { get; set; }
        public int SeatId { get; set; }
        public User User { get; set; } = null!;
        public Seat Seat { get; set; } = null!;
        public DateTime BookingDateTime { get; set; }
        public Booking()
        {
                UserId = string.Empty;
        }
        public Booking(int id, string userId, int seatId, DateTime bookingDateTime)
        {
            Id = id;
            UserId = userId;
            SeatId = seatId;
            BookingDateTime = bookingDateTime;
        }
    }
}